dali/kernels/alloc.cc

```diff
@@ -15,7 +15,11 @@
 #include <cuda_runtime.h>
 #include <cassert>
 #include "dali/kernels/alloc.h"
+#if !defined(__AARCH64_GNU__) && !defined(__AARCH64_QNX__)
 #include "dali/kernels/static_switch.h"
```

Comment:
Is `boost/preprocessor` that problematic for your build system? It's just preprocessor, no library dependencies there.

Reply:
Yes, boost was creating issues for QNX cross-compilation

Reply:
These headers are not present in aarch64 platforms:
https://github.com/NVIDIA/DALI/blob/master/dali/kernels/static_switch.h#L52-L54

Reply:
Isn't boost license is permissive enough to allow us to bundle boost/preprocessor headers with DALI?

Reply:
I don't see such headers present in either toolchains or DALI

Reply:
we are downloading boost source code in our Dockerfile.deps. Can't we do the same in the aarch64 version of it?

Reply:
What about QNX?

Reply:
@jantonguirao: What about QNX?

Reply:
@andi4191 Could you explain the limitations you faced with QNX cross compilation?

Reply:
@jantonguirao @mzient : fixed the boost compilation issue via Dockerfile